### DISCUSSION TITLE
Using "A" quickmark conflicts with the default "A" binding ("bmark")

### DISCUSSION THREAD

**Discussion by vadz** [tridactyl/tridactyl#4742] [Q&A] [answered]
I've used "A" to quickmark a site, but using e.g. <kbd>gnA</kbd> doesn't open this site in a new tab but bookmarks the current page instead. Is it possible to make this work while still preserving the original "A" function (i.e. by only suppressing it after `gn` and similar) or is `unbind A` the only way to use "A" as a quickmark?

Thanks for any ideas!

**Comment by bovine3dom** [comment]
The binds don't conflict for me:

```
:quickmark A
gnA # page opens in new tab
A # bookmark added
```

Could you post your config? `:viewconfig --user nmaps` :)

**Reply by vadz** [reply]
First of all, thanks for the lightning fast reply!

My full nmaps contains a bunch of other quickmarks with personal sites, so I'd rather not include it here, but they all have the same form as the one shown here, i.e. just `gnX`, `goX` and `gwX` mappings for different letters (both upper and lower-case ones, including one for `a`) but nothing else:
```json
{
  "gnA":"tabopen https://calendar.google.com/calendar/u/0/r?pli=1",
  "goA":"open https://calendar.google.com/calendar/u/0/r?pli=1",
  "gwA":"winopen https://calendar.google.com/calendar/u/0/r?pli=1"
}
```

After `unbind A` I also get `"A":null` there and then things work fine.

**Reply by bovine3dom** [reply]
Is it replicable for other keys, e.g. a quickmark to B conflicting with :taball?

**Reply by vadz** [reply]
No, I actually already use `B` for another quickmark and it works fine (and typing just `B` works fine too, although I never use it).

FWIW I'm using Firefox 102.13 ESR from Debian 12 (Bookworm) under Wayland with the latest version of Tridactyl as I've just installed this system a few days ago.

**Reply by bovine3dom** [reply, marked as answer]
That's deeply suspicious :)

I think there must be something strange going on in your config. If you try in a new Firefox profile do you see the same behaviour? `firefox -P` to bring up the profile menu.

If that fixes it I would recommend using `:mkt --clipboard` to store your current config somewhere, deleting everything from it that you do not need or recognise, then using `:sanitise tridactyllocal` to wipe your settings. Reimport with `:source --url` or look at this page to import from the clipboard https://github.com/tridactyl/tridactyl/wiki/import-export-tridactylrc-with-clipboard

**Reply by vadz** [reply]
Thanks a lot for the hint, just dumping the config and looking at it showed that, somehow, it had:

1. Multiple (identical) definitions of `g{n,o,w}X` commands.
2. Definitions for all the standard key bindings, i.e. it also had all this:
```
bind <A-p> pin
bind <A-m> mute toggle
bind <F1> help
bind o fillcmdline open
bind O current_url open
...
bind A bmark
...
bind B fillcmdline taball
...
bind <AC-`> mode ignore
bind <AS-ArrowUp><AS-ArrowUp><AS-ArrowDown><AS-ArrowDown><AS-ArrowLeft><AS-ArrowRight><AS-ArrowLeft><AS-ArrowRight>ba open https://www.youtube.com/watch?v=M3iOROuTuMA
```

I have no idea about how did this happen, but removing everything but the (single copy of) quickmark bindings definitions, sanitizing and restoring fixed the problem.

Thanks again!

**Reply by bovine3dom** [reply]
You're welcome :)

The default config getting copied in is a known issue - #4336 - there is a bug where sometimes interacting with the help page causes it to happen.
